### ISSUE TITLE
Update isodatetime

### ISSUE THREAD

**Issue by MetRonnie** [cylc/cylc-flow#4110]
**Describe exactly what you would like to see in an upcoming release**

Update the version of isodatetime used in cylc-flow once https://github.com/metomi/isodatetime/issues/176 is done and 2.1 is released. This should give performance improvements for datetime cycling workflows (see https://github.com/metomi/isodatetime/issues/162). Also, if https://github.com/metomi/isodatetime/issues/80 gets fixed that vanquishes a long-standing bug that is in cylc-flow.

*Optimistically tagged against 8.0rc1 as it would be good to have done before then, but not a priority. See discussion below.*

**Pull requests welcome!**

**Comment by MetRonnie** [comment]
@oliver-sanders @hjoliver I have this tagged against 8.x, but seeing as there is a breaking change (https://github.com/metomi/isodatetime/pull/183), would this be be better suited to 8.0.0?

**Comment by oliver-sanders** [comment]
The breaking change shouldn't be a problem (Cylc is used as a system rather than a library), good to keep 8.0.0 to the more mission-critical aspects right now.

**Comment by MetRonnie** [comment]
Forgot to mention, we could just release isodatetime 2.1.0 now and finish metomi/isodatetime#176 later in 2.1.1. Upgrading cylc-flow to isodatetime 2.1.0 shouldn't take too long.

**Comment by MetRonnie** [comment]
Optimistically tagged against 8.0.0 as it would be good to have done before then, but not a priority
